src/psyclone/tests/psyGen_test.py

```diff
@@ -2716,8 +2716,124 @@ def test_codeblock_gen_c_code():
+    assert coloredtext+"[annotations='was_elseif']" in output
+
+
+def test_ifblock_can_be_printed():
+    '''Test that an IfBlock instance can always be printed (i.e. is
+    initialised fully)'''
+    ifblock = IfBlock()
+    ref1 = Reference('condition1', parent=ifblock)
+    ifblock.addchild(ref1)
+    sch = Schedule(parent=ifblock)
+    ifblock.addchild(sch)
+    ret = Return(parent=sch)
+    sch.addchild(ret)
+
+    assert "If[]\n" in str(ifblock)
+    assert "condition1" in str(ifblock)  # Test condition is printed
+    assert "Return[]" in str(ifblock)  # Test if_body is printed
+
+
+def test_ifblock_properties():
+    '''Test that an IfBlock node properties can be retrieved'''
+    ifblock = IfBlock()
+
+    # Condition can't be retrieved before is added as a children.
+    with pytest.raises(InternalError) as err:
+        _ = ifblock.condition
+    assert("IfBlock malformed or incomplete. It should have "
+           "at least 1 children, but found 0." in str(err.value))
+
+    ref1 = Reference('condition1', parent=ifblock)
+    ifblock.addchild(ref1)
+
+    # If_body can't be retrieved before is added as a children.
+    with pytest.raises(InternalError) as err:
+        _ = ifblock.if_body
+    assert("IfBlock malformed or incomplete. It should have "
+           "at least 2 children, but found 1." in str(err.value))
+
+    sch = Schedule(parent=ifblock)
+    ifblock.addchild(sch)
+    ret = Return(parent=sch)
+    sch.addchild(ret)
+
+    # Now we can retrieve the condition and the if_body, but else is empty
+    assert ifblock.condition is ref1
+    assert ifblock.if_body[0] is ret
+    assert not ifblock.else_body
+
+    sch2 = Schedule(parent=ifblock)
+    ifblock.addchild(sch2)
+    ret2 = Return(parent=sch2)
+    sch2.addchild(ret2)
+
+    # Now we can retrieve else_body
+    assert ifblock.else_body[0] is ret2
+
+
+def test_ifblock_gen_c_code():
+    '''Test that an IfBlock node can generate its C representation'''
+
+    ifblock = IfBlock()
+    with pytest.raises(GenerationError) as err:
+        _ = ifblock.gen_c_code()
+    assert("IfBlock malformed or incomplete. It should have "
+           "at least 2 children, but found 0." in str(err.value))
+
+    ref1 = Reference('condition1', parent=ifblock)
+    ifblock.addchild(ref1)
+    with pytest.raises(GenerationError) as err:
+        _ = ifblock.gen_c_code()
+    assert("IfBlock malformed or incomplete. It should have "
+           "at least 2 children, but found 1." in str(err.value))
+
+    sch = Schedule(parent=ifblock)
+    ifblock.addchild(sch)
+    ret = Return(parent=sch)
+    sch.addchild(ret)
+
+    assert "if (condition1) {\n" in ifblock.gen_c_code()
+    assert "return;\n" in ifblock.gen_c_code()
+    assert "}" in ifblock.gen_c_code()
+    assert "else" not in ifblock.gen_c_code()
+
+    sch = Schedule(parent=ifblock)
+    ifblock.addchild(sch)
+    ret = Return(parent=sch)
+    sch.addchild(ret)
+    assert "else" in ifblock.gen_c_code()
```

Comment:
Please could you make this last check a little more thorough to check that the correct form is being produced.

Reply:
We may also want to start thinking about adding support for compiling generated C code as part of the test suite...please could you create a new Issue for this.

Reply:
Now full if-else expression is checked.

Reply:
Created issue #372